### DISCUSSION TITLE
^M problem when launching NeoVim

### DISCUSSION THREAD

**Discussion by ikkihou** [ayamir/nvimdots#722] [Q&A] [answered]
- Platform: WSL2 Ubuntu20.04
- NeoVim version: v0. 9. 0

hi guys, I am new to Neovim and  when configuring it with this repo, I got this error below, there are several `^M` errors:
![image](https://user-images.githubusercontent.com/51254965/236722797-bfce8fe3-1248-4a56-b572-d5ef7dbf48f6.png)

## Solutions I have tried
1. **dos2unix** the error file, however, it doesn't seem to take effect 

I went through the entire Issues tab and didn't find any similar conditions, so I post this issue.

**Comment by charliie-dev** [comment]
it seems like u have a poor internet connection (maybe over proxy?).

try deleting the plugin folder, reopen nvim again and let the plugin manager reinstall everything again.

**Reply by ikkihou** [reply]
thanks for your reply! I tried what you suggested, but unfortunately, it didn't work, I still got the same error. also, I indeed have a proxy via clash for windows.

 by the way, every plugin is installed successfully.

![image](https://user-images.githubusercontent.com/51254965/236733077-70480fe8-aa94-4951-bbab-342f3addd968.png)

**Reply by ikkihou** [reply]
![image](https://user-images.githubusercontent.com/51254965/236733237-3aa1f92e-294a-415d-b9ff-0908681e3ec3.png)

**Reply by charliie-dev** [reply]
hmmm, so the plugins look fully installed.

can u run ':checkhealth' and show the results?

**Reply by charliie-dev** [reply]
btw how do u install your neovim? cause i ran fine on ubuntu22.04 on wsl2 before.

**Reply by ikkihou** [reply]
I ran `:checkhealth`  and got 3 **warnings** listed below:
![Screenshot 2023-05-08 131228](https://user-images.githubusercontent.com/51254965/236739650-cf55627f-7f89-413f-a20d-0ac00385c09f.png)
![Screenshot 2023-05-08 131250](https://user-images.githubusercontent.com/51254965/236739654-f17e48dd-004d-4d5e-9b3d-6f7112c3ebe4.png)

**Reply by ikkihou** [reply]
I installed my neovim from the official github neovim repo: https://github.com/neovim/neovim/releases

and I used **linux 64x appimage** to install my neovim

**Reply by charliie-dev** [reply]
hmmm, those warnings are fine to ignore.

**Reply by charliie-dev** [reply]
Can u tried the `tar.gz` version instead of the appimage? I'm not sure if appimages work fine in wsl2.

**Comment by ayamir** [comment]
https://github.com/vim-autoformat/vim-autoformat/issues/355#issuecomment-864980911

**Reply by ikkihou** [reply, marked as answer]
appreciate your help!
I solved the problem by the two steps below:
### step 1
```
git config --global core.autocrlf false
```
### step 2 add the line below to ~/.vimrc (if not exists then create one)
```
set formatfile=unix
```

Finally, I delete all the plugins and the ~/.config/nvim folder and clone the **nvimdots** repo, and run nvim, then everything goes well. Lol.

I guess the main cause of this E488,E492 error is the `core.autocrlf` setting in git config.
